utils/firestore.js

```diff
@@ -3013,13 +3013,12 @@ const getSpecimensByCollectionIds = async (collectionIdsArray, siteCode, isBPTL
 }
 
 const processSendGridEvent = async (event) => {
-    if (event.gcloud_project !== process.env.GCLOUD_PROJECT) return;
+    if (!event.notification_id || event.gcloud_project !== process.env.GCLOUD_PROJECT) return;
 
     const date = new Date(event.timestamp * 1000).toISOString();
```

Comment:
This `date` is derived from `event.timestamp`. Do we need both `date` and `timestamp` in line 2028~2029, or `date` as a ISO time string is sufficient?

Reply:
Solved